Amazon.QLDB.Driver.Tests/AsyncTransactionTests.cs

```diff
@@ -0,0 +1,165 @@
+        }
+        
+        [TestMethod]
+        [ExpectedException(typeof(OccConflictException), AllowDerivedTypes = false)]
+        public async Task TestCommitWhenOCCShouldThrowException()
+        {
+            mockSession.Setup(s => s.CommitTransactionAsync(It.IsAny<string>(), It.IsAny<MemoryStream>(),
+                It.IsAny<CancellationToken>())).Throws(new OccConflictException(It.IsAny<string>()));
+
+            await asyncTransaction.Commit();
+            mockSession.Verify(s => s.AbortTransactionAsync(It.IsAny<CancellationToken>()), Times.Never);
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(AmazonServiceException), AllowDerivedTypes = false)]
+        public async Task TestCommitWhenAmazonServiceExceptionShouldThrowException()
+        {
+            mockSession.Setup(s => s.CommitTransactionAsync(It.IsAny<string>(), It.IsAny<MemoryStream>(),
+                It.IsAny<CancellationToken>())).Throws(new AmazonServiceException(It.IsAny<string>()));
+
+            await asyncTransaction.Commit();
+            mockSession.Verify(s => s.AbortTransactionAsync(It.IsAny<CancellationToken>()), Times.Exactly(1));
+        }
+        
+        [TestMethod]
+        [ExpectedException(typeof(InvalidOperationException), AllowDerivedTypes = false)]
+        public async Task TestCommitWhenDifferentTxnIDThrowsException()
+        {
+            mockSession.Setup(s => s.CommitTransactionAsync(It.IsAny<string>(), It.IsAny<MemoryStream>(),
+                It.IsAny<CancellationToken>())).Returns(GetTransactionResult("differentTnxIdFromThis"));
+
+            await asyncTransaction.Commit();
+        }
+        
+        [TestMethod]
+        public async Task TestExecuteCallsExecuteStatement()
+        {
+            mockSession.Setup(s => s.ExecuteStatementAsync(It.IsAny<string>(), It.IsAny<string>(),
+                It.IsAny<List<IIonValue>>(), It.IsAny<CancellationToken>()))
+                .ReturnsAsync(new ExecuteStatementResult
+                {
+                    FirstPage = new Page
+                    {
+                        NextPageToken = "nextPageToken",
+                        Values = new List<ValueHolder>()
+                    }
+                });
+
+
+            var result = await asyncTransaction.Execute("statement");
+
+            mockSession.Verify(s => s.ExecuteStatementAsync(TxnId, "statement", It.IsAny<List<IIonValue>>(),
+                It.IsAny<CancellationToken>()));
+            Assert.IsNotNull(result);
+        }
+     
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentException), AllowDerivedTypes = false)]
+        public async Task TestExecuteWhenStatementEmptyThrowsException()
+        {
+            await asyncTransaction.Execute("");
+        }
+
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
+        internal static async Task<CommitTransactionResult> GetTransactionResult(string txnId)
```

Comment:
Does this need to return `Task<CommitTransactionResult>`? 
Can you try adding `await` to the above lambda instances of `s.CommitTransactionAsync()` so that `Returns(GetTransactionResult()` will accept `CommitTransactionResult`?

If that works then lets keep `GetTransactionResult()` defined in `TransactionTests.cs` and use that function in `AsyncTransactionTests.cs` rather than the other way around.

Reply:
This method is to mock the return value of `[CommitTransactionAsync](https://github.com/justing-bq/amazon-qldb-driver-dotnet/blob/async-driver-dev/Amazon.QLDB.Driver/client/Session.cs#L201)` which returns a Task<CommitTransactionResult>, so it must return the same type